src/server/scripts/Kalimdor/Firelands/boss_ragnaros_firelands.cpp

```diff
@@ -0,0 +1,5304 @@
+                DoCastAOE(SPELL_ACHIEVEMENT_CHECK, true);
+                DoCastAOE(SPELL_HEART_OF_RAGNAROS_A, true);
+                DoCastAOE(SPELL_HEART_OF_RAGNAROS_H, true);
+                DoCastAOE(SPELL_AWARD_REPUTATION, true);
+            }
+
+            void EnterEvadeMode(EvadeReason why) override
+            {
+                if (!_EnterEvadeModeRagnaros(why))
+                    return;
+
+                _resetDueToEvade = true;
+                events.Reset();
+                CombatCleanup();
+                PreparePhase(PHASE_NONE);
+
+                if (instance)
+                {
+                    instance->SetBossState(DATA_RAGNAROS, FAIL);
+                    instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+                    if (GameObject* platform = ObjectAccessor::GetGameObject(*me, instance->GetGuidData(GO_RAGNAROS_PLATFORM)))
+                        platform->SetDestructibleState(GO_DESTRUCTIBLE_INTACT);
+
+                    // Only despawn if we find instance
+                    me->DespawnOrUnsummon();
+                }
+            }
+
+            bool _EnterEvadeModeRagnaros(EvadeReason why)
+            {
+                if (!me->IsAlive())
+                    return false;
+
+                if (why == EVADE_REASON_NO_HOSTILES
+                    && events.IsInPhase(PHASE_SUBMERGE)
+                    && !me->GetMap()->IsHeroic())
+                    return false;
+
+                //me->RemoveAurasOnEvade();
+                RemoveAuras(RagnarosAuraRemovePredicate(me));
+
+                // sometimes bosses stuck in combat?
+                me->DeleteThreatList();
+                me->CombatStop(true);
+                me->LoadCreaturesAddon();
+                me->SetLootRecipient(nullptr);
+                me->ResetPlayerDamageReq();
+                me->SetLastDamagedTime(0);
+
+                if (me->IsInEvadeMode())
+                    return false;
+
+                return true;
+            }
+
+            template <class Predicate>
+            void RemoveAuras(Predicate& predicate)
+            {
+                for (std::multimap<uint32, AuraApplication*>::iterator iter = me->GetAppliedAuras().begin(); iter != me->GetAppliedAuras().end();)
+                {
+                    SpellInfo const* spell = iter->second->GetBase()->GetSpellInfo();
+                    if (predicate(spell->Id))
+                        me->RemoveAura(iter);
+                    else
+                        ++iter;
+                }
+            }
+
+            void CombatCleanup(bool wipe = true)
+            {
+                if (me->GetMap()->IsHeroic() && wipe)
+                {
+                    summons.DespawnEntry(NPC_CENARIUS);
+                    summons.DespawnEntry(NPC_HAMUUL_RUNETOTEM);
+                    summons.DespawnEntry(NPC_MALFURION_STORMRAGE);
+                }
+
+                summons.DespawnEntry(NPC_CLOUDBURST);
```

Comment:
is there any temporary summon that should not be despawned? maybe it would be better to reverse the logic and despawn everything except that thing? (may need to extend SummonList)